src/core/Akka.Remote/Endpoint.cs

```diff
@@ -185,26 +187,29 @@ public ShutDownAssociationException(Address localAddress, Address remoteAddress,
         public Address RemoteAddress { get; private set; }
     }
 
-    internal sealed class InvalidAddressAssociationException : EndpointException, IAssociationProblem
+    internal sealed class InvalidAssociation : EndpointException, IAssociationProblem
```

Comment:
Changed the name of this exception to match the JVM (addressed some earlier comments by @rogeralsing)